HeatStroke/GOComponents/Components/ComponentSphereCollider.cpp

```diff
@@ -0,0 +1,133 @@
+//------------------------------------------------------------------------
+// ComponentSphereCollider
+// Author:	David MacIntosh
+//	
+// Collider that takes the shape of a sphere
+//------------------------------------------------------------------------
+
+#include "ComponentSphereCollider.h"
+
+#include "Collisions\CollisionManager.h"
+
+HeatStroke::ComponentSphereCollider::ComponentSphereCollider(
+	HeatStroke::GameObject* p_pGameObject,
+	glm::vec3 p_pOffset,
+	float p_fRadius)
+	:
+	ComponentCollider(p_pGameObject),
+	m_pOffset(p_pOffset),
+	m_fRadius(p_fRadius)
+{
+	HeatStroke::CollisionManager::Instance()->RegisterCollider(this, p_pGameObject->GetGUID());
+}
+
+HeatStroke::ComponentSphereCollider::~ComponentSphereCollider()
+{
+}
+
+HeatStroke::Component* HeatStroke::ComponentSphereCollider::CreateComponent(
+	HeatStroke::GameObject* p_pGameObject,
+	tinyxml2::XMLNode* p_pBaseNode,
+	tinyxml2::XMLNode* p_pOverrideNode)
+{
+	// Make some assertions on the parameters.
+	assert(p_pGameObject != nullptr);
+
+	// The values we need to fill by the end of parsing.
+	glm::vec3 offset;
+	float radius;
+
+	// Neither base nor override node are mandatory, but it is mandatory
+	// that all strings are given values between the two of them.
+
+	// Parse the elements of the base node.
+	if (p_pBaseNode != nullptr)
+	{
+		//ParseNode(p_pBaseNode, strOBJFileName);
+		ParseNode(p_pBaseNode, offset, radius);
+	}
+	// Then override with the Override node.
+	if (p_pOverrideNode != nullptr)
+	{
+		//ParseNode(p_pOverrideNode, strOBJFileName);
+		ParseNode(p_pOverrideNode, offset, radius);
+	}
+
+	// Check that we got everything we needed.
+	//assert(strOBJFileName != "");
+
+	// Now we can create and return the Component.
+	return new ComponentSphereCollider(
+		p_pGameObject,
+		offset,
+		radius
+		);
+}
+
+void HeatStroke::ComponentSphereCollider::Init()
+{
+}
+
+void HeatStroke::ComponentSphereCollider::Update(const float p_fDelta)
+{
+}
+
+void HeatStroke::ComponentSphereCollider::SyncTransform()
+{
+}
+
+void HeatStroke::ComponentSphereCollider::ParseNode(
+	tinyxml2::XMLNode* p_pNode,
+	glm::vec3& p_pOffset,
+	float& p_fRadius)
+{
+	assert(p_pNode != nullptr);
+	assert(strcmp(p_pNode->Value(), "GOC_SphereCollider") == 0);
+
+	for (tinyxml2::XMLNode* pChildNode = p_pNode->FirstChild();
```

Comment:
Same question I've asked others: is there a particular reason you're using `FirstChild()` instead of `FirstChildElement()`?  Would prevent the call to `ToElement()` later on, and remove the need for the check for comments.